From f82c2e8c43e11312075d851fde489d35e3b0a070 Mon Sep 17 00:00:00 2001
From: Murali Karicheri <m-karicheri2@ti.com>
Date: Mon, 11 Dec 2017 16:39:19 -0500
Subject: [PATCH 118/120] prueth: fix debugfs directory to include pruss id

Current implementation of debugfs interface for hsr/prp breaks the
driver when the second instance of the driver is brought up on ICSS1.
This is because the path for hsr/prp debugfs directory doesn't take
into account the possibility of having a second instance. To allow
second instance and to keep backward compatibility with other
applications such as net-snmp that assumes the path, create the
ICSS2 instance with prueth-prp path and the ICSS1 instance with a
prueth-prp-0 path. This ensures existing application works fine.

Beware that currently the prueth driver works fine for eth4, but
not for eth5. As the SAN - PTP uses PRP in emac mode and BC requires
at least one Eth port functional, this should be fine.

Signed-off-by: Murali Karicheri <m-karicheri2@ti.com>
---
 drivers/net/ethernet/ti/prueth.c | 19 ++++++++++++++++---
 1 file changed, 16 insertions(+), 3 deletions(-)

diff --git a/drivers/net/ethernet/ti/prueth.c b/drivers/net/ethernet/ti/prueth.c
index b843de4..45f7ba9 100644
--- a/drivers/net/ethernet/ti/prueth.c
+++ b/drivers/net/ethernet/ti/prueth.c
@@ -1257,18 +1257,31 @@ int prueth_hsr_prp_debugfs_init(struct prueth *prueth)
 	struct device *dev = prueth->dev;
 	int rc = -1;
 	struct dentry *de = NULL;
+	int id = prueth->pruss_id;
+	char dir[32];
+
+	memset(dir, 0, sizeof(dir));
+	if (prueth->fw_data->driver_data == PRUSS_AM57XX)
+		id -= 1;
 
 	if (PRUETH_HAS_HSR(prueth)) {
-		de = debugfs_create_dir("prueth-hsr", NULL);
+		if (id == 1)
+			sprintf(dir, "prueth-hsr");
+		else
+			sprintf(dir, "prueth-hsr%d", id);
 	} else if (PRUETH_HAS_PRP(prueth)) {
-		de = debugfs_create_dir("prueth-prp", NULL);
+		if (id == 1)
+			sprintf(dir, "prueth-prp");
+		else
+			sprintf(dir, "prueth-prp%d", id);
 	} else {
 		dev_err(dev, "unknown eth_type: %u\n", prueth->eth_type);
 		return -EINVAL;
 	}
 
+	de = debugfs_create_dir(dir, NULL);
 	if (!de) {
-		dev_err(dev, "Cannot create hsr-prp debugfs root\n");
+		dev_err(dev, "Cannot create %s debugfs root\n", dir);
 		return rc;
 	}
 
-- 
1.9.1

